### ISSUE TITLE
Menu with flipping dropdown does not remove aria-hidden on close

### ISSUE THREAD

**Issue by billop** [mantinedev/mantine#3515]
### What package has an issue

@mantine/core

### Describe the bug

When using a Menu that is set up so that the drop down "flips", closing the menu does not remove all the `aria-hidden` attributes throughout the page, leaving the page inaccessible.

I think this has something to do with use-focus-trap, but I'm not sure why it fails when the dropdown flips.

Currently working around this by adding
```ts
onClose={() =>
  Array.from(document.querySelectorAll('body > :not(script)')).map((node) =>
    node.removeAttribute('aria-hidden')
  )}
```
to my Menu.

### What version of @mantine/hooks page do you have in package.json?

5.10.3

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/frosty-tdd-s46ff0

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by DanB-dev** [comment]
I have proposed a fix for this.

**Comment by rtivital** [comment]
Fixed in 5.10.4
